MyCryptoHQ/MyCrypto
# Node Refactor (Pt. 4 - Electron Network Selector)

Closes #1455, closes #1353, and closes #1598.

### Description

**The final node refactor!** Implements the same network selector seen in #1640 by making the network selector component fill its content, and using it in both the header and electron nav. The only iffy part of this PR is that I do some conditional variable-ing to adjust certain minor styles between Electron and Web. Most of the time I'd make it a component param, but in this case, I only want it to render one way on web, and another on Electron, so I figured this was a workable case for that.

### Changes

* Broke out `Header`s `NetworkDropdown` into a standalone `NetworkSelector` component
* Prevent switching to the same node if clicked again (Handled in both component and saga)
* Added some additional accessibility titles to the network selector
* Minor type adjustments

### Steps to Test

1. Run the web build
2. Confirm the network selector behaves normally
3. Run the Electron build
4. Confirm the network selector behaves like web

### Screenshots

<img width="1332" alt="screen shot 2018-04-23 at 1 43 59 pm" src="https://user-images.githubusercontent.com/649992/39143606-640d4404-46fc-11e8-82b8-7f04512129bf.png">
